test/e2e/common.go

```diff
@@ -17,11 +17,55 @@ limitations under the License.
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/bootstrap"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	capiutil "sigs.k8s.io/cluster-api/util"
 
 	. "github.com/onsi/ginkgo" // nolint:gci
 )
 
+const (
+	KubernetesVersion    = "KUBERNETES_VERSION"
+	VsphereStoragePolicy = "VSPHERE_STORAGE_POLICY"
+)
+
+// Test suite global vars
+var (
+	// e2eConfig to be used for this test, read from configPath.
+	e2eConfig *clusterctl.E2EConfig
+
+	// clusterctlConfigPath to be used for this test, created by generating a clusterctl local repository
+	// with the providers specified in the configPath.
+	clusterctlConfigPath string
+
+	// bootstrapClusterProvider manages provisioning of the bootstrap cluster to be used for the e2e tests.
+	// Please note that provisioning will be skipped if e2e.use-existing-cluster is provided.
+	bootstrapClusterProvider bootstrap.ClusterProvider
+
+	// bootstrapClusterProxy allows to interact with the bootstrap cluster to be used for the e2e tests.
+	bootstrapClusterProxy framework.ClusterProxy
+
+	namespaces map[*corev1.Namespace]context.CancelFunc
+)
+
 func Byf(format string, a ...interface{}) {
 	By(fmt.Sprintf(format, a...))
 }
+
+func setupSpecNamespace(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string) (*corev1.Namespace, context.CancelFunc) {
```

Comment:
Not necessary to bring this change in, use the CAPI E2E code as a library. Similar to the comment I have added below.